[Yang Zhenzhao] iP

Zhenzha0 opened the pull request.

Zhenzha0 committed:
level 0

Zhenzha0 committed:
Level-1

Zhenzha0 committed:
Level-2

Zhenzha0 committed:
Level-3

lukeai-tan commented on src/main/java/Zuke.java:
```diff
@@ -37,16 +38,65 @@ public static void main(String[] args) {
+                if (counter == 0) {
+                    error("Your list is empty. Add a task first.");
+                    continue;
                 }
```
You could consider abstracting each command into its own method to give rise to better SLAP.

lukeai-tan commented on src/main/java/Zuke.java:
```diff
@@ -60,4 +110,36 @@ public static void main(String[] args) {
         System.out.println("____________________________________________________________");
     }
+
+    private static void error(String msg) {
+        printLine();
+        System.out.println("Error: " + msg);
+        printLine();
+    }
```
Nice abstraction for the error function. Keeps error formatting consistent throughout the code.

lukeai-tan reviewed (commented):
Overall LGTM. Code is easy to follow, just some duplication and some logic to abstract out so it'll be easier to maintain as the program grows.

Zhenzha0 committed:
refactor: extract Task class and split CLI into separate classes

- Added Task class to represent tasks with done state
- Introduced TaskList for managing task collection
- Created Ui for printing and Parser for input
- Simplified Main to handle only the event loop

Kart04 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,47 @@
+            return null;
+        }
+        try {
+            int idx1 = Integer.parseInt(text);
```
Perhaps you could use index1 instead of indx1

Kart04 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+public class Task {
+    protected String description;
+    protected boolean done;
```
Perhaps you could use isDone as this a boolean variable.

Kart04 commented on src/main/java/Zuke.java:
```diff
@@ -0,0 +1,53 @@
+        TaskList tasks = new TaskList();
+        Scanner in = new Scanner(System.in);
+
+        while (true) {
```
I like how you used a switch statement for the different cases instead of nested if statements, which helps avoid the arrow-shaped indentation and improves readability. The flat structure makes it easier to follow and maintain.

Kart04 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,57 @@
+public class Ui {
```
I like how you have a clear use of consistent naming conventions and PascalCase class

lukeai-tan commented on src/main/java/Zuke.java:
```diff
@@ -0,0 +1,53 @@
+                Ui.bye();
+                return;
+
+            case LIST:
+                if (tasks.isEmpty()) {
+                    Ui.error("Your list is empty. Add a task first.");
+                } else {
+                    Ui.showList(tasks);
+                }
+                break;
+
+            case MARK: {
+                if (tasks.isEmpty()) { Ui.error("Your list is empty. Add a task first."); break; }
+                Integer idx = Parser.parseIndexOrNull(c.arg, tasks.size());
+                if (idx == null) break;
+                tasks.mark(idx);
+                Ui.showMarked(tasks.get(idx), true);
+                break;
+            }
+
+            case UNMARK: {
+                if (tasks.isEmpty()) { Ui.error("Your list is empty. Add a task first."); break; }
+                Integer idx = Parser.parseIndexOrNull(c.arg, tasks.size());
+                if (idx == null) break;
+                tasks.unmark(idx);
+                Ui.showMarked(tasks.get(idx), false);
+                break;
+            }
+
+            case ADD:
+                tasks.add(c.arg);                 // c.arg == original line
+                Ui.showAdded(c.arg);
+                break;
+            }
```
You could consider abstracting the code for each case into their own separate functions to promote SLAP.

lukeai-tan replied:
You could consider abstracting out the code for each case and making it their own functions.

lukeai-tan commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,57 @@
+public class Ui {
+    public static void line() {
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void logo() {
+        String logo = "$$$$$$$$\\ $$\\   $$\\ $$\\   $$\\ $$$$$$$$\\ \n" +
+                "\\____$$  |$$ |  $$ |$$ | $$  |$$  _____|\n" +
+                "    $$  / $$ |  $$ |$$ |$$  / $$ |      \n" +
+                "   $$  /  $$ |  $$ |$$$$$  /  $$$$$\\    \n" +
+                "  $$  /   $$ |  $$ |$$  $$<   $$  __|   \n" +
+                " $$  /    $$ |  $$ |$$ |\\$$\\  $$ |      \n" +
+                "$$$$$$$$\\ \\$$$$$$  |$$ | \\$$\\ $$$$$$$$\\ \n" +
+                "\\________| \\______/ \\__|  \\__|\\________|\n";
+        System.out.println(logo);
+    }
+
+    public static void hello() {
+        line();
+        System.out.println("Hello! I'm Zuke\nWhat can I do for you?");
+        line();
+    }
+
+    public static void bye() {
+        line();
+        System.out.println("Bye. Hope to see you again soon!");
+        line();
+    }
+
+    public static void error(String msg) {
+        line();
+        System.out.println("Error: " + msg);
+        line();
+    }
```
I like how you abstracted out all the dialogue code into their separate functions. Makes the code more readable!

lukeai-tan replied:
Well done on making a class for all the Ui commands. Makes the code more readable and intuitive.

lukeai-tan commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,47 @@
+public class Parser {
```
Great job on creating a Parser class to handle inputs. This helps promote SRP in your code!

lukeai-tan replied:
Great job on adding a parser to handle all inputs. Really helps with promoting SRP in your code.

lukeai-tan commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,47 @@
+
+    public enum Type { LIST, MARK, UNMARK, ADD, BYE }
+
+    public static class Command {
+        public final Type type;
+        public final String arg;   // for ADD or index text for MARK/UNMARK
+
+        public Command(Type type, String arg) {
+            this.type = type;
+            this.arg = arg;
+        }
+    }
```
LGTM. You could consider making this a class outside of Parser, so that you can create other Commands to inherit from it in the future.

lukeai-tan replied:
You could perhaps put the Command class in a separate file. The way Command class is implemented is very clean though. LGTM.

lukeai-tan reviewed (commented):
Overall LGTM! Very good SLAP and SRP principles utilised. Just a couple of code to abstract out such as the ones in the switch cases and you should be all good!

lukeai-tan commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,57 @@
+public class Ui {
+    public static void line() {
+        System.out.println("____________________________________________________________");
+    }
+
```
Good job on abstracting the line printing!

lukeai-tan reviewed (commented):
Overall LGTM. Your code is very well structured and readable. Just some minor changes you could consider, mainly the abstraction stuff. Good job!

Zhenzha0 committed:
Level-4 without error handling

Added event, todo and deadline subclasses
Added event and deadline parser to parse arguements
Changed UI output implementation for adding tasks

Zhenzha0 committed:
Level-4

Added error handling for different command and argument errors

Zhenzha0 committed:
Error handling

Added error handling for different command and argument errors

Zhenzha0 committed:
Level-4

added showTaskSize function to in Ui

Zhenzha0 committed:
resolved merge

Zhenzha0 committed:
Final Level-4 for real

added missed out error handling for todo

Zhenzha0 committed:
Text Ui Testing

Zhenzha0 committed:
Ammended expected text

Zhenzha0 committed:
Improved error handling using try-catch

Zhenzha0 committed:
Level-5

Merge branch 'branch-Level-5'

Zhenzha0 committed:
A-Packages

Zhenzha0 committed:
Refractored try-catch statements

lukeai-tan commented on src/main/java/Task/Parser/EventParser.java:
```diff
@@ -0,0 +1,60 @@
+
+
+
+    public static String checkArgumentFormat(String description, String from , String to) {
```
```suggestion
    public static String checkArgumentFormat(String description, String from, String to) {

```

lukeai-tan commented on src/main/java/Task/parser/EventParser.java:
```diff
@@ -0,0 +1,60 @@
+                continue;
+            }
+
+            if(currentFlag.equals("description")){
+                descriptionBuilder.append(words[i]);
+                descriptionBuilder.append(" ");
+            } else if(currentFlag.equals("from")){
+                fromBuilder.append(words[i]);
+                fromBuilder.append(" ");
+            } else if(currentFlag.equals("to")){
+                toBuilder.append(words[i]);
+                toBuilder.append(" ");
+            }
```
```suggestion
            if (currentFlag.equals("description")) {
                descriptionBuilder.append(words[i]);
                descriptionBuilder.append(" ");
            } else if (currentFlag.equals("from")) {
                fromBuilder.append(words[i]);
                fromBuilder.append(" ");
            } else if (currentFlag.equals("to")) {
                toBuilder.append(words[i]);
                toBuilder.append(" ");
            }

```

lukeai-tan commented on src/main/java/Zuke.java:
```diff
@@ -0,0 +1,125 @@
+
+            } catch (ZukeException.UnknowInputException | ZukeException.MissingArgumentException | ZukeException.EmptyListException | ZukeException.MissingDescriptionException e) {
+                Ui.error(e.getMessage());
+            }
```
```suggestion
            } catch (ZukeException e) {
                Ui.error(e.getMessage());
            }

```

lukeai-tan commented on src/main/java/Task/TaskList.java:
```diff
@@ -0,0 +1,44 @@
+        tasks.get(zeroBasedIndex).unmark();
+    }
+
+    public String render() {
+        StringBuilder sb = new StringBuilder("Here are the tasks in your list:\n");
+        for (int i = 0; i < tasks.size(); i++) {
+            sb.append(i + 1).append(". ").append(tasks.get(i)).append('\n');
+        }
+        return sb.toString();
+    }
```
Very elegant use of StringBuilder instead of string concatenation for efficiency. Well done!

lukeai-tan commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,34 @@
+
+public class Task {
+    protected String description;
+    protected boolean done;
```
You could consider naming this isDone to follow the standard boolean naming convention.

lukeai-tan reviewed (commented):
Overall LGTM. There are some minor coding standard issues to iron out such as line length, and missing spaces for if-else, for, and while loops. Very good job on promoting SLAP and SRP principles for the most part.

namsengi11 commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,68 @@
+        line();
+    }
+
+    public static void showMarked(Task t, boolean nowDone) {
```
t could be renamed to task to be more descriptive.

namsengi11 commented on src/main/java/exception/ZukeException.java:
```diff
@@ -0,0 +1,50 @@
+package Exception;
+
+public class ZukeException {
```
Ideally, each class representing an exception should be defined in separate files.

namsengi11 commented on src/main/java/Task/Parser/CommandParser.java:
```diff
@@ -0,0 +1,59 @@
+
+public class CommandParser {
+
+    public enum Type { LIST, MARK, UNMARK, BYE, TODO, DEADLINE, EVENT, UNKNOWN }
```
Good job using enums to make code more readable.

namsengi11 commented on src/main/java/Command/CommandParser.java:
```diff
@@ -0,0 +1,59 @@
+        String rest = parts.length > 1 ? parts[1] : null;
+
+        switch (cmd) {
+        case "list":   return new Command(Type.LIST, null);
```
You could divide each case and corresponding code into separate lines to make flow more readable, instead of using one-liners.

namsengi11 commented on src/main/java/Command/CommandParser.java:
```diff
@@ -0,0 +1,59 @@
+                throw new ZukeException.MissingIndexException();
+            }
+
+            int idx1 = Integer.parseInt(text);
```
You could rename the variable to be more descriptive, as it isn't immediately obvious what idx1 represents.

Zhenzha0 committed:
Level-6

Zhenzha0 committed:
added text file data base

Zhenzha0 committed:
Level-7

Zhenzha0 committed:
Level-7 improved

Zhenzha0 committed:
Merge branch 'branch-Level-6'

Zhenzha0 committed:
Merge branch 'branch-Level-7'

Zhenzha0 committed:
more oop

Zhenzha0 committed:
even more oop

Zhenzha0 committed:
More OOP!!!

Zhenzha0 committed:
Implementetd Level 8

Added DateTimeParser class
refractored deadline and event class
refractored deadline and event parser classes
refractored classes in Memory package
refractored exception handling in commandparser

Zhenzha0 committed:
Added searcher package with datesearcher and description searcher classes

Zhenzha0 committed:
Integrated searchers into commandhander class

Zhenzha0 committed:
corrected mistake at find and date switch case

Zhenzha0 committed:
Merge pull request #1 from Zhenzha0/branch-Level-9

Branch level 9

Zhenzha0 committed:
Added JavaDoc comments for all classes and methods

Zhenzha0 committed:
Merge pull request #2 from Zhenzha0/branch-A-JavaDoc

Zhenzha0 committed:
fixed storage issue on missing directory

Zhenzha0 committed:
Updated testing texts and updated find error handling

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Create _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Create assets

Zhenzha0 committed:
Delete docs/assets

Zhenzha0 committed:
Add files via upload

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
Update _config.yml

Zhenzha0 committed:
changed class and package naming according to naming convention

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
removed extra ui line at welcome

Zhenzha0 committed:
Merge branch 'master' of https://github.com/Zhenzha0/ip

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
deleted data file for github page

Zhenzha0 committed:
Update README.md

Zhenzha0 committed:
Fix image path in README.md
